### ISSUE TITLE
Restructure macro-related exports into `private` mods

### ISSUE THREAD

**Issue by ggwpez** [paritytech/substrate#14213]
Check all crates that define macros and put the macro-related reexports into a `private` module that will be excempt from semver.

**Comment by thiolliere** [comment]
to find where they are one can grep for `#[doc(hidden)]` flags in the codebase

**Comment by kianenigma** [comment]
Is this a reasonable first step toward breaking down frame-support in its entirety? if so, it'd be good to have a parent tracking issue as well. 

I think other than macros, the rest of frame support is easy to break apart, and mostly self explanatory.

**Comment by juangirini** [comment]
> Is this a reasonable first step toward breaking down frame-support in its entirety? if so, it'd be good to have a parent tracking issue as well.
> 
> I think other than macros, the rest of frame support is easy to break apart, and mostly self explanatory.

Good point, I have added it to https://github.com/paritytech/substrate/issues/14777
